src/VegaChart.tsx

```diff
@@ -69,9 +71,23 @@ export const VegaChart: FC<VegaChartProps> = ({
 		return { [TABLE]: clonedData };
 	}, [data]);
 
-	const previousChartData = usePreviousChartData({ [PREVIOUS_TABLE]: chartData.table });
+	const previousChartData = useMemo(() => {
+		const clonedData = JSON.parse(JSON.stringify(previousData));
```

Comment:
Could we use a true deep copy here?  We can import `structuredClone` from elsewhere in the project